pyomo/gdp/tests/models.py

```diff
@@ -162,29 +162,34 @@ def makeTwoTermDisjOnBlock():
     m = ConcreteModel()
     m.b = Block()
     m.a = Var(bounds=(0,5))
-    def d_rule(disjunct, flag):
+
+    # On a whim, verify that the decorator notation works
+    @m.b.Disjunct([0,1])
```

Comment:
Where is the decorator defined? Is this relying on the same decorator as `Constraint`?

Reply:
Yes - this makes use of the standard decorator functionality.  The `@model.Constraint(...)` notation works for any component that takes a rule as a keyword argument.